src/form/components/Form.tsx

```diff
@@ -120,6 +126,12 @@ export const Form = <FormData extends {}>({
     })
   }, [hiddenFormFields])
 
+  useEffect(() => {
+    changedRepeatFields.forEach((repeatSection) => {
+      internalOnChange(repeatSection.field.lens, repeatSection.value)
+    })
+  }, [changedRepeatFields])
```

Comment:
why not just run a useEffect on the [visibleFormFields]. and then compare the repeat fields groups length to its value, and update/slice if not consistent?

Reply:
I also need to update the non-visible fields (e.g. if I set the flag doRedeemDebts to false which then sets the length of the numberOfDebts to null and in another round adjusts the list to an empty list ... That is why it was so tricky.

Reply:
I feel like we should just ax repeat form field groups, and then this problem becomes so much easier

Reply:
nvm, we use it. i thought we only used the section one